models/vyos/vyos_firewall/vyos_firewall.yaml

```diff
@@ -0,0 +1,212 @@
+              type: dict
+              suboptions:
+                sending:
+                  description:
+                    - Permits or denies transmitting packets ICMP redirect messages.
+                  type: bool
+                receiving:
+                  description:
+                    - Permits or denies receiving packets ICMP redirect messages.
+                  type: bool
+        ping:
+          description:
+            - Policy for handling of all IPv4 ICMP echo requests.
+          type: dict
+          suboptions:
+            all:
+              description:
+                - Enables or disables response to all IPv4 ICMP Echo Request (ping) messages.
+                - The system responds to IPv4 ICMP Echo Request messages.
+              type: bool
+            broadcast:
+              description:
+                - Enables or disables response to broadcast IPv4 ICMP Echo Request and Timestamp Request messages.
+                - IPv4 ICMP Echo and Timestamp Request messages are not processed.
+              type: bool
+        config_trap:
+          description:
+            - SNMP trap generation on firewall configuration changes.
+          type: bool
+        group:
+          description:
+            - Defines a group of objects for referencing in firewall rules.
+          type: dict
+          suboptions:
+             address_group:
+               description:
+                 - Defines a group of IP addresses for referencing in firewall rules.
+               type: list
+               elements: dict
+               suboptions:
+                 name:
+                   description:
+                     - Name of the firewall address group.
+                   type: str
+                   required: True
+                 description:
+                   description:
+                     - Allows you to specify a brief description for the address group.
+                   type: str
+                 address:
+                   description:
+                     - Address-group members.
+                     - IPv4 address to match.
+                     - IPv4 range to match.
+                   type: list
+             network_group:
+               description:
+                 - Defines a group of networks for referencing in firewall rules.
+               type: list
+               elements: dict
+               suboptions:
+                 name:
+                   description:
+                     - Name of the firewall network group.
+                   type: str
+                 description:
+                   description:
+                     - Allows you to specify a brief description for the network group.
+                   type: str
+                 network:
+                   description:
+                     - Adds an IPv4 network to the specified network group.
+                     - The format is ip-address/prefix.
```

Comment:
@ganeshrn  should i keep it as network or change it to address ?

Reply:
@rohitthakur2590 address seems in consistent with the approach other platform follows, so it's better to have address